src/generate_form.jsx

```diff
@@ -0,0 +1,306 @@
+    }
+
+    function compute() {
+        var calc_sum = 0;
+        var err_check = 0;
+        var newErrorState = Array.from({length: props.fields_info.length}, (v, k) => false);
+
+        for (var i = 0; i < props.fields_info.length; i++) {
+            const field = props.fields_info[i];
+            const val = (field.type == "Float" ? parseFloat(props.getValue(field.name)) : parseInt(props.getValue(field.name)));
+            if (!props.isSomeValueEntered(field.name) || isNaN(val)) {
+                err_check = -1;
+                newErrorState[i] = true;
+            } else {
+                newErrorState[i] = false;
+            }
+            calc_sum += val * field.weight;
+        }
+
+        if (err_check == 0) {
+            setSum(calc_sum);
+            setOpen(true);
+        }
+        // Feel free to add any error reporting via: 
+        // setErrDialog({show: true, message: "Your message here"});
+        props.setErrorState(newErrorState);
+    }
+
+    return (
+        <div>
+            <Button variant="contained" color="primary" className={classes.button} onClick={compute}>Compute HCC Score</Button>
+            <Dialog
+                maxWidth='xl'
+                open={open}
+                onClose={handleClose}>
+                <DialogContent>
+                    <Typography variant="overline" display="block" gutterBottom>HCC RECURRENCE SCORE</Typography>
+                    <Typography variant="h1" component="h2" gutterBottom>
+                        {(Math.round(sum * Math.pow(10, config.roundoff)) / Math.pow(10, config.roundoff)).toString()+"%"}
+                    </Typography>
+                </DialogContent>
+            </Dialog>
+            <Dialog open={errDialog.show} onClose={handleErrDialogClose}>
+                <DialogContent>
+                    <Typography variant="body1" gutterBottom>{errDialog.message}</Typography>
+                </DialogContent>
+            </Dialog>
+        </div>
+    );
+}
+
+function Info_button() {
+    const tooltip_classes = useTooltipStyles();
+    return (
+        <Tooltip 
+            enterTouchDelay={0}
+            title={config.info}
+            placement="bottom-start"
+            classes={tooltip_classes}>
+            <InfoIcon />
+        </Tooltip>);
+}
+
+class Fields extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            errorState: Array.from({length: props.fields_info.length}, (v, k) => false),
+            values: {}
+        };
+    }
+
+    render() {
+        return (
+            <Grid container direction="column" alignItems="center">
+            <Grid item>
+
+            <Grid container direction="column" justify="space-between" alignItems="flex-start" spacing={3}>
+
+              <Grid item>
+                  <Grid container>
+                    <Grid item xs={"auto"}>
+                        <Typography variant="h3" gutterBottom>{config.title}</Typography>
```

Comment:
You may want this to look like an `h3`, but semantically it should be a `h1` as it's the first (and only) title on the page. Use the `component` attribute to generate an `h1` element that will look like a `h3` as specified by the `variant` attribute.

```suggestion
                        <Typography variant="h3" component="h1" gutterBottom>{config.title}</Typography>
```